src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -143,11 +143,14 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
   // We will also register ourselves as CaptureStatusListener, in order to get notified on
   // capture finish (OnCaptureFinished). That's when we will reset the open query slots.
   void SetVulkanLayerProducer(VulkanLayerProducer* vulkan_layer_producer) {
+    // De-register from the previous VulkanLayerProducer.
```

Comment:
Do you know if de-registering was also the intent of the previous code? It seems like it but I'm not 100% sure if that was the reason for `vulkan_layer_producer->SetCaptureStatusListener(nullptr)`.

Reply:
I belief it was intended, but I am totally unsure, why? Is it because we can handle multiple devices/instances (in theory)?

Reply:
I would guess it was mainly for completeness.

Reply:
Probably.